src/base/projects/BranchSelector.svelte

```diff
@@ -0,0 +1,123 @@
+<script lang="ts">
+  import { getBranchesByCommit, Info } from "@app/project";
+  import { formatCommit } from "@app/utils";
+
+  export let branches: [string, string][];
+  export let project: Info;
+  export let selectedBranch: [string, string] | null;
```

Comment:
The `selectedBranch` information here is persisted in the `View` component and gets passed through the `Header` component until here, this is to allow being persistent while the Header and the BranchSelector get rerendered with changing commits.